modules/server/src/network/resolvers/aggregations.ts

```diff
@@ -1,239 +1,85 @@
-			return { ...acc, [requestedFieldKey]: field };
-		} else {
-			return acc;
-		}
-	}, {});
-
-	const gqlString = createNodeQueryString(documentName, fieldsToRequest);
-
-	/*
-	 * convert string to AST object
-	 * not needed if gqlString is formatted correctly but this acts as a validity check
-	 */
-	try {
-		const gqlQuery = gql`
-			${gqlString}
-		`;
-		return gqlQuery;
-	} catch (err) {
-		console.error('invalid gql', err);
-		return undefined;
-	}
+import { RequestedFieldsMap } from '../utils/gql';
+import { fetchData } from './fetch';
+import { createNetworkQuery } from './query';
+
+export const CONNECTION_STATUS = {
+	OK: 'OK',
+	ERROR: 'ERROR',
+} as const;
+
+export type NetworkNode = {
+	name: string;
+	hits: number;
+	status: keyof typeof CONNECTION_STATUS;
+	errors: string;
+	aggregations: { name: string; type: string }[];
 };
 
-type SuccessResponse = { [k: string]: { hits: Hits; aggregations: AllAggregations } };
+type SuccessResponse = Record<string, { hits: Hits; aggregations: AllAggregationsMap }>;
 
 /**
  * Query each remote connection
  *
  * @param queries - Query for each remote connection
- * @param requestedAggregationFields
- * @returns
+ * @param requestedAggregationFields - Fields requested
+ * @returns Resolved aggregation and node info
  */
 export const aggregationPipeline = async (
 	configs: NodeConfig[],
 	requestedAggregationFields: RequestedFieldsMap,
-	queryVariables: QueryVariables,
+	queryVariables: AggregationsQueryVariables,
 ) => {
 	const nodeInfo: NetworkNode[] = [];
 
 	const totalAgg = new AggregationAccumulator(requestedAggregationFields);
 
-	const aggregationResultPromises = configs.map(async (config) => {
-		const gqlQuery = createNetworkQuery(config, requestedAggregationFields);
-		const response = gqlQuery
-			? await fetchData<SuccessResponse>({
-					url: config.graphqlUrl,
-					gqlQuery,
-					queryVariables,
-			  })
-			: failure(CONNECTION_STATUS.ERROR, 'Invalid GQL query');
-
-		const nodeName = config.displayName;
-
-		if (isSuccess(response)) {
-			const documentName = config.documentName;
-			const responseData = response.data[documentName];
-			const aggregations = responseData?.aggregations || {};
-			const hits = responseData?.hits || { total: 0 };
-
-			totalAgg.resolve({ aggregations, hits });
-
-			nodeInfo.push({
-				name: nodeName,
-				hits: hits.total,
-				status: CONNECTION_STATUS.OK,
-				errors: '',
-				aggregations: config.aggregations,
-			});
-		} else {
-			nodeInfo.push({
-				name: nodeName,
-				hits: 0,
-				status: CONNECTION_STATUS.ERROR,
-				errors: response?.message || 'Error',
-				aggregations: config.aggregations,
-			});
-		}
-	});
+	await Promise.allSettled(
+		configs.map(async (config) => {
+			// create node query
+			const gqlQuery = createNetworkQuery(config, requestedAggregationFields);
+
+			// query node
+			const response = gqlQuery
+				? await fetchData<SuccessResponse>({
```

Comment:
is this an instance of the fetchGql function I saw elsewhere? how are those two different? 🤔 
I'm starting to get confused here ("me" problem, cognitive load)

Reply:
hm i see now this one was the function that summoned the other one where I left that earlier comment...
though it's obvious I need more coffee, I think less generic / more verbose variable names could help keep track of things as one adds more code into mental context

Reply:
it's a bit overloaded cognitively for sure!
the `fetchGQL` is just an axios call with configured request options - headers, query vars
this `fetchData` calls `fetchGQL` - formats the GQL query string for the requests, handles response errors, returns success/failure objects

Reply:
sounds like a good reason confirming the need for me to abstract all these into a single shared set of functions later on, once this is all merged